### PR TITLE
[BEAM-13747] Add e2e test for BigQuery Write API beam connector

### PR DESCRIPTION
**Pull request by yirutang**
Add a e2e test for BigQuery Write API connector.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by yirutang** [comment]
Run PostCommit_Java_Dataflow

**Comment by yirutang** [comment]
Run PostCommit_Java_Dataflow

**Comment by yirutang** [comment]
R: @reuvenlax

**Comment by yirutang** [comment]
Run PostCommit_Java_Dataflow

**Comment by yirutang** [comment]
R: @chamikaramj

**Comment by yirutang** [comment]
Run Java_PreCommit

**Comment by yirutang** [comment]
Run Java PreCommit

**Comment by yirutang** [comment]
Run Java PreCommit

**Comment by yirutang** [comment]
Run PostCommit_Java_Dataflow

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                    new TableFieldSchema().setName("number").setType("INTEGER"),
+                    new TableFieldSchema().setName("str").setType("STRING")));
+
+    Pipeline p = Pipeline.create(bqOptions);
+    p.apply("Input", GenerateSequence.from(0).to(rowCount))
+        .apply("GenerateMessage", ParDo.of(new FillRowFn()))
+        .apply(
+            "WriteToBQ",
+            BigQueryIO.writeTableRows()
+                .to(String.format("%s:%s.%s", project, DATASET_ID, tableName))
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java:98]: Looking at the job executions, seems like tests ran using BATCH_LOADS mode for some reason.
https://pantheon.corp.google.com/dataflow/jobs/us-central1/2022-03-09_17_22_34-16871784130096288394;step=?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))

https://pantheon.corp.google.com/dataflow/jobs/us-central1/2022-03-09_17_22_34-16871784130096288394?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))


Can you try setting following ?

"withMethod(BigQueryIO.Write.Method.STORAGE_WRITE_API)" or "withMethod(BigQueryIO.Write.Method.STORAGE_API_AT_LEAST_ONCE).

**Review by chamikaramj — COMMENTED** [review]
Thanks.

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  private void setUpTestEnvironment(WriteMode writeMode) {
+    PipelineOptionsFactory.register(BigQueryOptions.class);
+    bqOptions = TestPipeline.testingPipelineOptions().as(BigQueryOptions.class);
+    bqOptions.setProject(TestPipeline.testingPipelineOptions().as(GcpOptions.class).getProject());
+    bqOptions.setUseStorageWriteApi(true);
+    if (writeMode == WriteMode.AT_LEAST_ONCE) {
+      bqOptions.setUseStorageWriteApiAtLeastOnce(true);
+    }
+    bqOptions.setNumStorageWriteApiStreams(2);
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java:70]: I think options "setNumStorageWriteApiStreams" and  "setStorageWriteApiTriggeringFrequencySec" are only intended for unbounded input (in streaming pipelines). @reuvenlax should this have resulted in a validation failure ?

**Review by chamikaramj — COMMENTED** [review]
Thanks.

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by chamikaramj** [comment]
Seems like pre-commit failures are related.

* What went wrong:
14:56:10 Execution failed for task ':sdks:java:io:google-cloud-platform:checkstyleTest'.
14:56:10 > Checkstyle rule violations were found. See the report at: file:///home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Phrase@2/src/sdks/java/io/google-cloud-platform/build/reports/checkstyle/test.html
14:56:10   Checkstyle files with violations: 1
14:56:10   Checkstyle violations by severity: [error:1]

**Comment by yirutang** [comment]
Run Java PreCommit

**Comment by yirutang** [comment]
Run PostCommit_Java_Dataflow

**Comment by chamikaramj** [comment]
Java post-commit passed previously and latest failure is unrelated.

**Comment by chamikaramj** [comment]
LGTM

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java (hunk 2 above)
Comment (chamikaramj) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java:70]: Let's address this seperately.

**Comment by kennknowles** [comment]
Java post-commit actually started failing at this PR

**Comment by kennknowles** [comment]
https://issues.apache.org/jira/browse/BEAM-14149

**Pull request merged**